src/app/products/item-card/item-card.component.ts

```diff
@@ -11,17 +11,17 @@ import { ProductData } from '../../services/product-data';
   styleUrl: './item-card.component.scss',
 })
 export class ItemCardComponent {
-  @Input() product!: ProductData;
-  @Output() productSelectDetail = new EventEmitter<ProductData>();
-  @Output() productAddToCart = new EventEmitter<ProductData>();
+  @Input() public product!: ProductData;
+  @Output() public productSelectDetail = new EventEmitter<ProductData>();
+  @Output() public productAddToCart = new EventEmitter<ProductData>();
 
   constructor() {}
 
-  onViewDetails(): void {
+  public onViewDetails(): void {
     this.productSelectDetail.emit(this.product);
   }
 
-  onProductSelectedToCart(): void {
+  public onProductSelectedToCart(): void {
     this.productAddToCart.emit(this.product);
   }
 }
```

Comment:
The same data type is emitted twice, can this be refactored into this ?
@Output() public productAction = new EventEmitter<{ type: 'detail' | 'addToCart', data: ProductData }>();

private emitProductAction(type: 'detail' | 'addToCart'): void {
  this.productAction.emit({ type, data: this.product });
}

public onViewDetails(): void {
  this.emitProductAction('detail');
}

public onProductSelectedToCart(): void {
  this.emitProductAction('addToCart');
}